t/03-testapi.t

```diff
@@ -231,6 +235,11 @@ subtest 'script_run' => sub {
 
     is(assert_script_run('true'), undef, 'nothing happens on success');
     $fake_exit = 1;
+
+    exception {assert_script_run ('false', timeout => 2, max_interval => 1000), 'false', "It failed"} or diag "The test failed";
+    diag explain $cmds;
+    BAIL_OUT("Testing is done");
+
```

Comment:
t/03-testapi.t .. 
ok 1 - isotovideo version set (variable is considered part of test API)
# Subtest: type_string
[2020-05-19T15:07:03.000 CEST] [debug] <<< testapi::type_string(string="hallo", max_interval=250, wait_screen_changes=0, wait_still_screen=0, timeout=30, similarity_level=47)
[2020-05-19T15:07:03.000 CEST] [debug] backend_type_string
    1..0
    # No tests run!
not ok 2 - No tests run for subtest "type_string"

#   Failed test 'No tests run for subtest "type_string"'
#   at t/03-testapi.t line 166.
Not a HASH reference at /home/jora/Git/os-autoinst-distri-opensuse/os-autoinst/autotest.pm line 331.
ok 3 - no (unexpected) warnings (via END block)
# Tests were run but no plan was declared and done_testing() was not seen.
# Looks like your test exited with 255 just after 3.
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 1/3 subtests 

Test Summary Report
-------------------
t/03-testapi.t (Wstat: 65280 Tests: 3 Failed: 1)
  Failed test:  2
  Non-zero exit status: 255
  Parse errors: No plan found in TAP output
Files=1, Tests=3,  0 wallclock secs ( 0.01 usr  0.00 sys +  0.26 cusr  0.01 csys =  0.28 CPU)
Result: FAIL

shell returned 1

Reply:
@Martchus could you please give a hint for the unit testing?

Reply:
The syntax in L239 is broken. Take a close look at the arguments of `assert_script_run`. As first argument you pass `(…)` and then comma-separated after that `false` and "It failed". I recommend you take over the same style as the other calls in the same subtest, e.g. use

```perl
like(exception {assert_script_run 'false', timeout => 2, max_interval => 1000; }, qr/false/, 'foo');
```

Make sure you also test that the `max_interval` is properly passed through to the underlying command structure, e.g. use a similar format as further down in this test with `like($cmds->[1]->{text}, qr/; echo /)` but check for `max_interval` instead

Reply:
*facepalms*, @DrMullings we need more coffee :P. 

@okurz with this, the changes should be properly tested right? as what's being passed to distribution is being tested through the test api itself (Or do I need more coffe here?)

Reply:
Yes, I would say so. So if you add one call of `assert_script_run` with the right parameter and just check that the parameter is forwarded I consider this sufficient testing.

Reply:
I hope I got the syntax correctly now